modules/lang-painless/src/main/java/org/elasticsearch/painless/MethodWriter.java

```diff
@@ -262,6 +262,50 @@ public void writeToStrings() {
         }
     }
 
+    /** Writes a dynamic binary instruction: returnType, lhs, and rhs can be different */
+    public void writeDynamicBinaryInstruction(Location location, Type returnType, Type lhs, Type rhs, Operation operation) {
+        org.objectweb.asm.Type descriptor = org.objectweb.asm.Type.getMethodType(returnType.type, lhs.type, rhs.type);
```

Comment:
I don't plan to change this on the issue. Please look at the changes, it used a switch on invokedynamic before this PR. Its not my fault. Lets fix it separate.